hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/DevSources.java

```diff
@@ -0,0 +1,187 @@
+     * on the delay at that's partially driven by the Jet engine. In practice it will behave similar to fixedDelay.
+     *
+     * The source is not distributed it means only a single Jet instance will emit items.
+     *
+     * @param delay minimum delay between emitting
+     * @param timeUnit units of the delay
+     * @return source emitting with a specified minimum delay
+     */
+    public static StreamSource<Long> minimumDelay(long delay, TimeUnit timeUnit) {
+        return minimumDelay(delay, timeUnit, System::currentTimeMillis);
+    }
+
+    /**
+     * Streaming source which emit items at fixed rate. It attempts to compensate for various system hiccups so
+     * the rate over a period of time is constant.
+     *
+     * You have to provide your own item supplier.
+     * The source is not distributed it means only a single Jet instance will emit items.
+     *
+     * @param period period which to emit
+     * @param timeUnit units for period
+     * @param itemSupplier supplier of items to be emitted. It's called whenever a new item is about to be emitted
+     * @param <T> type of emitted items
+     * @return source emitting with a fixed rate
+     */
+    public static <T> StreamSource<T> fixedRate(long period, TimeUnit timeUnit, SupplierEx<T> itemSupplier) {
+        return triggerDrivenStreamSource(() -> new FixedRateTrigger(period, timeUnit), itemSupplier);
+    }
+
+    /**
+     * Streaming source which emits with specified minimum delay between items.
+     *
+     * It won't emit unless elapsed time since last emit is at least the specified delay. There is no upper bound
+     * on the delay at that's partially driven by the Jet engine. In practice it will behave similar to fixedDelay.
+     *
+     * You have to provide your own item supplier.
+     * The source is not distributed it means only a single Jet instance will emit items.
+     *
+     * @param delay delay minimum delay between emitting
+     * @param timeUnit units of the delay
+     * @param itemSupplier supplier of items to be emitted. It's called whenever a new item is about to be emitted
+     * @param <T> type of emitted items
+     * @return source emitting with a specified minimum delay
+     */
+    public static <T> StreamSource<T> minimumDelay(long delay, TimeUnit timeUnit, SupplierEx<T> itemSupplier) {
+        return triggerDrivenStreamSource(() -> new MinDelayTrigger(delay, timeUnit), itemSupplier);
+    }
+
+    /**
+     * Batch source producing specified elements.
+     *
+     * @param items values to emit
+     * @param <T> type of the element
+     * @return source emitting all specified items
+     */
+    public static <T> BatchSource<T> of(T... items) {
+        return fromIterator(() -> Arrays.asList(items).iterator());
+    }
+
+    private static <T> StreamSource<T> triggerDrivenStreamSource(SupplierEx<Trigger> triggerSupplier,
+                                                                 SupplierEx<T> itemSupplier) {
```

Comment:
Our convention is to put the parameter list on its own line when it doesn't fit all into one line.

```
private static <T> StreamSource<T> triggerDrivenStreamSource(
        SupplierEx<Trigger> triggerSupplier, SupplierEx<T> itemSupplier
) {
```

Reply:
Also, `@Nonnull` must be everywhere, on the return values as well.